goss-testing/scripts/data-validator.py

```diff
@@ -192,103 +206,98 @@ def are_hosts_sane(data, desired_keys):
 
 
 def validate_ntp(data):
+  """Check ntp data.
+  """
+
   err = 0
   if are_valid_ip_masks(data, ['ntp', 'allow']): err = 1
   if are_hosts_sane(data, ['ntp', 'peers']): err = 1
   if are_hosts_sane(data, ['ntp', 'servers']): err = 1
   return err
 
 
-def score_params(node_class):
-  err = 0
-
-  lengths = []
-  counts = []
-  for blob in node_class:
-    lengths.append(blob['params-length'])
-  
-  for i in lengths:
-    occ = lengths.count(i)
-    counts.append(occ)
-    for blob in node_class:
-      if i == blob['params-length']:
-        blob['occurrence'] = occ
-
-  baseline = max(counts)
-
-  searches = [ "bond=bond0:mgmt0,mgmt1",
-               "ifname=mgmt0",
-               "ip=mgmt0",
-               "ifname=mgmt1",
-               "ip=mgmt1",
-               "hostname=" ]
-
-  all_param_strings = []
-  for blob in node_class:
-    for k,v in blob.items():
-      if k == "params-stripd":
-        all_param_strings.append(v)
-    if blob['occurrence'] == baseline:
-      for search in searches:
-        if re.search(search, blob['params-orig']):
-          blob['valid'] = True
-        else:
-          blob['valid'] = False
-    else:
-      blob['valid'] = False
-
-  unique_param_strings = set(all_param_strings)
-  if len(unique_param_strings) > 1:
-    for blob in node_class:
-      for params in unique_param_strings:
-        if params == blob['params-stripd']:
-          blob['valid'] = False
-
-
-  for blob in node_class:
-    if blob['valid'] == False:
-      print_err(f"WARN: {blob['hostname']}: has boot params that differ from at least one other node\n{blob['params-orig']}\n")
-      err = 1
-
-  print_err("------------------------------------------------------------")
-
-  if err == 1:
-    return err
-
-
 def boot_params(data):
-  """Attempt to validate boot parameters from BSS
+  """Confirm that params required for booting are
+     present.
   """
-  worker = []
-  storage = []
-  management = []
-  err = 0
 
-  macreg = re.compile(r':([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}')
-  hostreg = re.compile(r'hostname=[a-zA-Z]*\-[a-zA-Z0-9]*')
-  httpreg = re.compile(r's=http://.*/')
+  err = 0
+  ncn_params = [
+	"biosdevname=1",
+	"console=tty0",
+	"console=ttyS0,115200",
+	"crashkernel=[0-9]*M",
+	"ds=nocloud-net;s=http://",
+	"hostname=",
+	"ifname=mgmt0",
+	"ifname=mgmt1",
+	"initrd=initrd.img.xz",
+	"iommu=pt",
+	"log_buf_len=1",
+	"metal.no-wipe=",
+	"metal.no-wipe=1",
+	"metal.server=http://",
```

Comment:
This will vary between `metal.server=http://` and `metal.server=s3://`

- `http` is going to be on the cmdline
- `s3` will be in BSS, but will show up as `http` on the cmdline because BSS swaps it out before plugging the cmdline into the node (so it'll show up as `s3` when comparing to BSS, despite looking like `http` on the cmdline)

Reply:
Eventually this could even vary between `https` and `s3`.

Checking for `s3` in BSS is a good test, but idk if this can be tested against basecamp.

Also this param should *not* exist in a disk cmdline, it should *only* exist in a netbooting cmdline.

Reply:
I'll just look for **metal.server=**